lib/spack/spack/modules/common.py

```diff
@@ -269,13 +269,39 @@ def read_module_indices():
 module_type_to_indices = read_module_indices()
 
 
-def upstream_module(spec, module_type):
-    indices = module_type_to_indices[module_type]
+def upstream_module(spec, module_type, _module_indices=None):
+    indices = _module_indices or module_type_to_indices[module_type]
```

Comment:
can `module_type_to_indices` be computed lazily here instead of at module scope?

Reply:
New implementation computes module indices in `spack.store` lazily.